cmd/jujud/bootstrap.go

```diff
@@ -427,10 +427,37 @@ func (c *BootstrapCommand) populateTools(st *state.State, env environs.Environ)
 			SHA256:  tools.SHA256,
 		}
 		logger.Debugf("Adding tools: %v", toolsVersion)
-		if err := storage.Add(bytes.NewReader(data), metadata); err != nil {
+		if err := toolstorage.Add(bytes.NewReader(data), metadata); err != nil {
 			return errors.Trace(err)
 		}
 	}
+
+	// Store the Juju GUI archive.
+	guistorage, err := st.GUIStorage()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer guistorage.Close()
+	gui, err := agenttools.ReadGUIArchive(dataDir)
+	if err != nil {
```

Comment:
Well, in the future all errors will be important here, the GUI should never be not found.